packages/presentation/lib/screen/home/home_page.dart

```diff
@@ -1,177 +1,159 @@
-          title: Text(widget.title),
-        ),
-        body: Center(
-          child: Padding(
-            padding: const EdgeInsets.symmetric(horizontal: 15.0),
-            child: Column(
-              mainAxisAlignment: MainAxisAlignment.center,
-              children: <Widget>[
-                BlocSelector<HomeBloc, HomeState, int?>(
-                  selector: (state) => state.generatedNumber,
-                  builder: (context, generatedNumber) {
-                    return generatedNumber != null
-                        ? Text('Random number: $generatedNumber')
-                        : const SizedBox();
-                  },
-                ),
-                const Text(
-                  'Try to predict generated number:',
-                  style: TextStyle(fontSize: 20),
-                ),
-                const SizedBox(height: 15),
-                TextFormField(
-                  keyboardType: TextInputType.number,
-                  maxLength: 1,
-                  controller: _controller,
-                  decoration: const InputDecoration(
-                    border: OutlineInputBorder(),
-                    labelText: 'Number?',
-                  ),
-                  onChanged: (value) {
-                    BlocProvider.of<HomeBloc>(context).add(
-                      ChangeSubmitButtonStateEvent(
-                          isSubmitButtonActive: value.isNotEmpty),
-                    );
-                  },
-                ),
-                const SizedBox(height: 15),
-                Row(
+    );
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return StreamBuilder(
+      stream: bloc.dataStream,
+      builder: (context, snapshot) {
+        if (snapshot.data != null) {
+          final data = snapshot.data as BlocState;
+          final HomeState? blocData = data.data;
+
+          if (blocData == null) {
+            return Container();
+          }
+
+          return Scaffold(
+            appBar: AppBar(
+              title: const Text('Generate numbers'),
+            ),
+            body: Center(
+              child: Padding(
+                padding: const EdgeInsets.symmetric(horizontal: 15.0),
+                child: Column(
                   mainAxisAlignment: MainAxisAlignment.center,
-                  children: [
-                    ElevatedButton(
-                        onPressed: () {
-                          BlocProvider.of<HomeBloc>(context)
-                              .add(GenerateNumberEvent());
-                        },
-                        child: const Text('Generate')),
-                    const SizedBox(
-                      width: 15,
+                  children: <Widget>[
+                    blocData.generatedNumber != null
+                        ? Text('Random number: ${blocData.generatedNumber}')
+                        : const SizedBox(),
```

Comment:
you can use just if without ternary operator
if(blocData.generatedNumber != null)
Text('Random number: ${blocData.generatedNumber}')

Reply:
corrected